### PR TITLE
Add x509(1) reference (1.1.0)

### PR DESCRIPTION
**Pull request by vdukhovni**

### CODE REVIEW

File: doc/crypto/X509_VERIFY_PARAM_set_flags.pod [hunk 1]
```diff
@@ -338,7 +338,8 @@ connections associated with an B<SSL_CTX> structure B<ctx>:
 L<X509_verify_cert(3)>,
 L<X509_check_host(3)>,
 L<X509_check_email(3)>,
-L<X509_check_ip(3)>
+L<X509_check_ip(3)>,
+L<x509(1)|x509(1)>
```
Comment (richsalz) [inline doc/crypto/X509_VERIFY_PARAM_set_flags.pod]: I should make doc-nits find this kind of construct :)

Comment (paulidale) [reply doc/crypto/X509_VERIFY_PARAM_set_flags.pod]: Definitely.

Comment (vdukhovni) [reply doc/crypto/X509_VERIFY_PARAM_set_flags.pod]: Where are we with this PR?

Comment (richsalz) [reply doc/crypto/X509_VERIFY_PARAM_set_flags.pod]: Needs to be fixed, L<a|a> is wrong.

Comment (vdukhovni) [reply doc/crypto/X509_VERIFY_PARAM_set_flags.pod]: It has plenty of precident:
```
$ cd doc; pcregrep -lrc 'L<([^>]+)\|\1>' .
./crypto/PEM_write_bio_CMS_stream.pod:5
./crypto/ASN1_OBJECT_new.pod:2
./crypto/BN_copy.pod:2
./crypto/BN_bn2bin.pod:4
./crypto/CMS_uncompress.pod:1
./crypto/EVP_PKEY_CTX_new.pod:1
./crypto/CMS_get0_RecipientInfos.pod:2
./crypto/i2d_PKCS7_bio_stream.pod:5
./crypto/RSA_check_key.pod:2
./crypto/ERR_load_crypto_strings.pod:1
./crypto/blowfish.pod:3
./crypto/PKCS12_parse.pod:2
./crypto/X509_VERIFY_PARAM_set_flags.pod.orig:4
./crypto/RSA_padding_add_PKCS1_type_1.pod:4
./crypto/OPENSSL_load_builtin_modules.pod:1
./crypto/d2i_DHparams.pod:2
./crypto/PKCS7_sign.pod:1
./crypto/SSLeay_version.pod:1
./crypto/DSA_dup_DH.pod:2
./crypto/EVP_PKEY_keygen.pod:7
./crypto/d2i_X509_CRL.pod:2
./crypto/EVP_OpenInit.pod:4
./crypto/DH_generate_key.pod:2
./crypto/EVP_PKEY_encrypt.pod:9
./crypto/i2d_CMS_bio_stream.pod:5
./crypto/DSA_get_ex_new_index.pod:1
./crypto/X509_NAME_print_ex.pod:2
./crypto/RSA_set_method.pod:2
./crypto/PKCS7_sign_add_signer.pod:2
./crypto/RSA_generate_key.pod:6
./crypto/ERR_print_errors.pod:4
./crypto/ERR_get_error.pod:4
./crypto/EC_KEY_new.pod:9
./crypto/des.pod:5
./crypto/EC_GROUP_new.pod:4
./crypto/BIO_f_buffer.pod:5
./crypto/X509_STORE_CTX_new.pod:2
./crypto/DSA_SIG_new.pod:3
./crypto/EVP_PKEY_print_private.pod:2
./crypto/CMS_add1_recipient_cert.pod:2
./crypto/ecdsa.pod:2
./crypto/des_modes.pod:2
./crypto/evp.pod:32
./crypto/DH_new.pod:4
./crypto/BN_generate_prime.pod:2
./crypto/ASN1_generate_nconf.pod:2
./crypto/hmac.pod:1
./crypto/DH_get_ex_new_index.pod:1
./crypto/DH_generate_parameters.pod:5
./crypto/dh.pod:8
./crypto/X509_verify_cert.pod:2
./crypto/SMIME_write_PKCS7.pod:3
./crypto/CMS_verify_receipt.pod:4
./crypto/rand.pod:9
./crypto/lh_stats.pod:1
./crypto/OpenSSL_add_all_algorithms.pod:2
./crypto/BIO_s_fd.pod:6
./crypto/RSA_public_encrypt.pod:3
./crypto/X509_STORE_set_verify_cb_func.pod:2
./crypto/ERR_clear_error.pod:1
./crypto/EC_GROUP_copy.pod:4
./crypto/X509_STORE_CTX_set_verify_cb.pod:3
./crypto/BIO_new_CMS.pod:2
./crypto/EVP_PKEY_verify.pod:6
./crypto/PKCS7_verify.pod:2
./crypto/d2i_RSAPublicKey.pod:2
./crypto/DSA_generate_parameters.pod:5
./crypto/bn.pod:18
./crypto/threads.pod:1
./crypto/RAND_cleanup.pod:1
./crypto/BN_add.pod:7
./crypto/OPENSSL_VERSION_NUMBER.pod:1
./crypto/CMS_compress.pod:1
./crypto/RAND_load_file.pod:1
./crypto/OBJ_nid2obj.pod:1
./crypto/rsa.pod:12
./crypto/lhash.pod:1
./crypto/CMS_get0_type.pod:1
./crypto/DH_set_method.pod:2
./crypto/CMS_sign_receipt.pod:3
./crypto/EVP_PKEY_new.pod:2
./crypto/RAND_set_rand_method.pod:1
./crypto/X509_STORE_CTX_get_ex_new_index.pod:1
./crypto/d2i_PKCS8PrivateKey.pod:4
./crypto/EVP_PKEY_cmp.pod:2
./crypto/ASN1_STRING_length.pod:1
./crypto/pem.pod:1
./crypto/EC_POINT_new.pod:3
./crypto/X509_VERIFY_PARAM_set_flags.pod:5
./crypto/engine.pod:1
./crypto/crypto.pod:15
./crypto/CMS_final.pod:2
./crypto/bn_internal.pod:2
./crypto/RSA_print.pod:1
./crypto/X509_check_host.pod:6
./crypto/CRYPTO_set_ex_data.pod:4
./crypto/PKCS12_create.pod:1
./crypto/CMS_add0_cert.pod:3
./crypto/CMS_get0_SignerInfos.pod:2
./crypto/d2i_X509_NAME.pod:2
./crypto/ERR_put_error.pod:2
./crypto/EC_GFp_simple_method.pod:7
./crypto/EVP_PKEY_decrypt.pod:6
./crypto/ui_compat.pod:1
./crypto/X509_NAME_add_entry_by_txt.pod:1
./crypto/BN_BLINDING_new.pod:1
./crypto/RAND_add.pod:4
./crypto/d2i_CMS_ContentInfo.pod:2
./crypto/EVP_PKEY_derive.pod:6
./crypto/DSA_new.pod:4
./crypto/RAND_bytes.pod:3
./crypto/CMS_verify.pod:2
./crypto/BN_swap.pod:1
./crypto/BIO_read.pod:3
./crypto/BN_num_bytes.pod:2
./crypto/X509_NAME_get_index_by_NID.pod:1
./crypto/BN_mod_inverse.pod:2
./crypto/BN_CTX_new.pod:5
./crypto/BN_zero.pod:1
./crypto/ERR_load_strings.pod:1
./crypto/EVP_EncryptInit.pod:1
./crypto/PEM_write_bio_PKCS7_stream.pod:5
./crypto/PKCS7_decrypt.pod:1
./crypto/ERR_set_mark.pod:1
./crypto/DH_size.pod:1
./crypto/x509.pod:13
./crypto/ERR_remove_state.pod:1
./crypto/d2i_X509_SIG.pod:2
./crypto/d2i_X509.pod:4
./crypto/RSA_size.pod:1
./crypto/EVP_PKEY_sign.pod:10
./crypto/BIO_s_file.pod:6
./crypto/BIO_s_accept.pod:1
./crypto/ASN1_STRING_new.pod:1
./crypto/BN_mod_mul_reciprocal.pod:4
./crypto/X509_NAME_ENTRY_get_object.pod:2
./crypto/CMS_get1_ReceiptRequest.pod:3
./crypto/ui.pod:2
./crypto/md5.pod:2
./crypto/DSA_do_sign.pod:5
./crypto/d2i_ECPKParameters.pod:4
./crypto/RSA_blinding_on.pod:1
./crypto/err.pod:16
./crypto/RSA_private_encrypt.pod:4
./crypto/mdc2.pod:2
./crypto/EVP_PKEY_verify_recover.pod:6
./crypto/DSA_size.pod:1
./crypto/EVP_SignInit.pod:7
./crypto/CMS_encrypt.pod:1
./crypto/RSA_new.pod:4
./crypto/DSA_set_method.pod:2
./crypto/RAND_egd.pod:5
./crypto/CONF_modules_load_file.pod:2
./crypto/SMIME_write_CMS.pod:3
./crypto/BN_rand.pod:3
./crypto/d2i_ASN1_OBJECT.pod:2
./crypto/OPENSSL_config.pod:2
./crypto/d2i_DSAPublicKey.pod:2
./crypto/bio.pod:13
./crypto/EVP_BytesToKey.pod:2
./crypto/PKCS7_encrypt.pod:1
./crypto/SMIME_read_PKCS7.pod:4
./crypto/d2i_ECPrivateKey.pod:10
./crypto/EVP_DigestVerifyInit.pod:6
./crypto/ripemd.pod:2
./crypto/RSA_sign.pod:6
./crypto/DSA_generate_key.pod:3
./crypto/BN_CTX_start.pod:3
./crypto/sha.pod:2
./crypto/ASN1_STRING_print_ex.pod:2
./crypto/EVP_VerifyInit.pod:8
./crypto/BIO_s_bio.pod:2
./crypto/d2i_X509_REQ.pod:2
./crypto/RSA_get_ex_new_index.pod:2
./crypto/EC_POINT_add.pod:4
./crypto/ec.pod:10
./crypto/EVP_PKEY_get_default_digest.pod:4
./crypto/EVP_PKEY_CTX_ctrl.pod:8
./crypto/SMIME_read_CMS.pod:4
./crypto/BN_mod_mul_montgomery.pod:4
./crypto/buffer.pod:1
./crypto/EVP_DigestInit.pod:2
./crypto/BN_set_bit.pod:2
./crypto/EVP_SealInit.pod:4
./crypto/EVP_DigestSignInit.pod:6
./crypto/dsa.pod:13
./crypto/BN_add_word.pod:2
./crypto/CMS_decrypt.pod:1
./crypto/CMS_sign.pod:1
./crypto/DSA_sign.pod:3
./crypto/d2i_X509_ALGOR.pod:2
./crypto/CONF_modules_free.pod:2
./crypto/ERR_error_string.pod:7
./crypto/EVP_PKEY_set1_RSA.pod:1
./crypto/ERR_GET_LIB.pod:1
./crypto/BIO_f_ssl.pod:1
./crypto/rc4.pod:2
./crypto/X509_new.pod:2
./crypto/RSA_sign_ASN1_OCTET_STRING.pod:4
./crypto/BN_cmp.pod:1
./crypto/X509_STORE_CTX_get_error.pod:1
./crypto/CMS_add1_signer.pod:2
./crypto/BN_new.pod:2
./ssl/SSL_CTX_set_cert_cb.pod:5
./ssl/SSL_set_fd.pod:3
./ssl/SSL_CTX_set_max_cert_list.pod:4
./ssl/SSL_CTX_set_cert_store.pod:6
./ssl/SSL_CONF_CTX_set_ssl_ctx.pod:5
./ssl/SSL_clear.pod:12
./ssl/SSL_CTX_free.pod:2
./ssl/SSL_CTX_add_session.pod:5
./ssl/SSL_get_ex_new_index.pod:7
./ssl/SSL_CTX_set_quiet_shutdown.pod:10
./ssl/SSL_SESSION_get_time.pod:4
./ssl/SSL_CONF_cmd.pod:5
./ssl/SSL_new.pod:4
./ssl/SSL_CTX_sess_number.pod:4
./ssl/SSL_CTX_get0_param.pod:1
./ssl/SSL_CTX_set_tmp_rsa_callback.pod:5
./ssl/SSL_CTX_set_read_ahead.pod:1
./ssl/SSL_get_fd.pod:1
./ssl/SSL_CTX_set_timeout.pod:9
./ssl/SSL_get_error.pod:1
./ssl/SSL_CTX_set_client_cert_cb.pod:10
./ssl/SSL_get_client_CA_list.pod:5
./ssl/SSL_CONF_CTX_set1_prefix.pod:5
./ssl/SSL_CTX_set_ssl_version.pod:7
./ssl/SSL_CTX_set_client_CA_list.pod:6
./ssl/SSL_get_ex_data_X509_STORE_CTX_idx.pod:3
./ssl/SSL_pending.pod:4
./ssl/SSL_set_verify_result.pod:4
./ssl/SSL_set_session.pod:5
./ssl/SSL_CTX_new.pod:2
./ssl/SSL_check_chain.pod:2
./ssl/SSL_get_current_cipher.pod:2
./ssl/SSL_CIPHER_get_name.pod:3
./ssl/SSL_CTX_set_session_id_context.pod:1
./ssl/SSL_CTX_set_cert_verify_callback.pod:6
./ssl/SSL_CTX_set_default_passwd_cb.pod:2
./ssl/SSL_get_ciphers.pod:3
./ssl/SSL_state_string.pod:1
./ssl/SSL_rstate_string.pod:1
./ssl/SSL_SESSION_free.pod:8
./ssl/SSL_CTX_get_ex_new_index.pod:5
./ssl/SSL_set_connect_state.pod:10
./ssl/SSL_get_version.pod:1
./ssl/SSL_get_verify_result.pod:5
./ssl/SSL_CTX_set1_verify_cert_store.pod:13
./ssl/SSL_CTX_add_extra_chain_cert.pod:15
./ssl/SSL_alert_type_string.pod:1
./ssl/SSL_load_client_CA_file.pod:3
./ssl/SSL_CONF_CTX_new.pod:5
./ssl/SSL_CTX_sess_set_get_cb.pod:10
./ssl/SSL_CTX_load_verify_locations.pod:9
./ssl/SSL_free.pod:5
./ssl/SSL_CTX_set1_curves.pod:1
./ssl/SSL_CONF_cmd_argv.pod:5
./ssl/SSL_CTX_set_info_callback.pod:4
./ssl/SSL_write.pod:12
./ssl/SSL_get_session.pod:7
./ssl/SSL_library_init.pod:2
./ssl/SSL_get_peer_certificate.pod:4
./ssl/SSL_CTX_use_certificate.pod:13
./ssl/SSL_CTX_set_options.pod:7
./ssl/SSL_CONF_CTX_set_flags.pod:5
./ssl/SSL_get_default_timeout.pod:6
./ssl/SSL_CTX_set_tlsext_ticket_key_cb.pod:6
./ssl/SSL_get_peer_cert_chain.pod:2
./ssl/SSL_CTX_set_mode.pod:2
./ssl/SSL_set_shutdown.pod:6
./ssl/SSL_CTX_set_session_cache_mode.pod:14
./ssl/SSL_get_SSL_CTX.pod:2
./ssl/SSL_connect.pod:5
./ssl/SSL_CTX_set_verify.pod:14
./ssl/SSL_CTX_add1_chain_cert.pod:1
./ssl/SSL_CTX_get_verify_mode.pod:1
./ssl/SSL_session_reused.pod:2
./ssl/SSL_get_rbio.pod:1
./ssl/SSL_set_bio.pod:3
./ssl/SSL_COMP_add_compression_method.pod:1
./ssl/SSL_CTX_set_msg_callback.pod:2
./ssl/SSL_CTX_flush_sessions.pod:6
./ssl/SSL_CTX_ctrl.pod:1
./ssl/SSL_accept.pod:5
./ssl/SSL_shutdown.pod:10
./ssl/ssl.pod:85
./ssl/SSL_CTX_sessions.pod:6
./ssl/SSL_CTX_set_tmp_dh_callback.pod:6
./ssl/SSL_read.pod:14
./ssl/SSL_want.pod:8
./ssl/SSL_CTX_sess_set_cache_size.pod:5
./ssl/SSL_do_handshake.pod:5
./ssl/SSL_SESSION_get_ex_new_index.pod:5
./ssl/SSL_CTX_set_cipher_list.pod:9
./ssl/SSL_CTX_set_generate_session_id.pod:2
./ssl/d2i_SSL_SESSION.pod:4
./apps/sess_id.pod:1
./apps/config.pod:1
./apps/dhparam.pod:2
./apps/pkeyparam.pod:2
./apps/pkcs12.pod:6
./apps/x509v3_config.pod:4
./apps/enc.pod:1
./apps/pkcs7.pod:1
./apps/tsget.pod:1
./apps/asn1parse.pod:2
./apps/rand.pod:2
./apps/gendsa.pod:3
./apps/s_client.pod:3
./apps/rsa.pod:4
./apps/crl.pod:2
./apps/ecparam.pod:2
./apps/errstr.pod:3
./apps/spkac.pod:2
./apps/ts.pod:16
./apps/dsaparam.pod:3
./apps/pkey.pod:4
./apps/dgst.pod:2
./apps/pkeyutl.pod:3
./apps/x509.pod:5
./apps/ca.pod:6
./apps/s_server.pod:3
./apps/req.pod:9
./apps/verify.pod:1
./apps/c_rehash.pod:3
./apps/genpkey.pod:1
./apps/smime.pod:2
./apps/CA.pl.pod:3
./apps/pkcs8.pod:4
./apps/cms.pod:3
./apps/crl2pkcs7.pod:1
./apps/ec.pod:3
./apps/rsautl.pod:1
./apps/dsa.pod:4
./apps/s_time.pod:7
./apps/ciphers.pod:1
./apps/openssl.pod:13
```

Comment (kaduk) [reply doc/crypto/X509_VERIFY_PARAM_set_flags.pod]: precedent could just mean that there is some existing code that has not been converted yet; it need not necessarily mean that we are obligated to accept new instances into the tree.

Comment (vdukhovni) [reply doc/crypto/X509_VERIFY_PARAM_set_flags.pod]: Indeed precedent alone is not sufficient, it's just that it was far from clear which is the correct or preferred syntax. I have a vague recollection that without explicit "a|b" forms, some references across manpage "sections" used to not work correctly.  So I don't have any preferences here, happy with any syntax that behaves correctly...

Comment (kaduk) [reply doc/crypto/X509_VERIFY_PARAM_set_flags.pod]: I am remembering some previous discussion about how a|a was redundant/unneeded and we should just use a, but it seems a hard thing to search for.  I would have expected @levitte to be involved in that thread, though on the other hand he did commit f0b843c1f4eade5f9d54f826b16cec5ebd15a502 which seems to not adhere to that doctrine (and be more recent than the discussion I think I remember).

Comment (richsalz) [reply doc/crypto/X509_VERIFY_PARAM_set_flags.pod]: In master I removed the redundancy; my mistake, not in 1.1.0
We should not add them back to master.  Previous releases are okay.

Comment (vdukhovni) [reply doc/crypto/X509_VERIFY_PARAM_set_flags.pod]: Oh, I see, my list was from 1.0.2, in master we have:
```
./crypto/X509_VERIFY_PARAM_set_flags.pod.orig:L<X509_verify_cert(3)|X509_verify_cert(3)>,
./crypto/X509_VERIFY_PARAM_set_flags.pod.orig:L<X509_check_host(3)|X509_check_host(3)>,
./crypto/X509_VERIFY_PARAM_set_flags.pod.orig:L<X509_check_email(3)|X509_check_email(3)>,
./crypto/X509_VERIFY_PARAM_set_flags.pod.orig:L<X509_check_ip(3)|X509_check_ip(3)>
```
which look like my additions from the X509 work around DANE... So, if the repetition is no longer useful, feel free to simplify.

**Comment by vdukhovni** [comment]
Updated to match 1.1.0 reference format.

**Comment by vdukhovni** [comment]
Merged and closed. Thanks

**Pull request closed without merging**